encoding/multi_reader_iterator.go

```diff
@@ -169,7 +169,7 @@ func (it *multiReaderIterator) moveIteratorToValidNext(iter Iterator) bool {
 	}
 
 	err := iter.Err()
-	if it.err == nil && err != nil {
+	if it.err == nil && err != nil && err != io.EOF {
```

Comment:
Interesting! I ran into this while working on tests for updating series' data. The test file in the PR replicates the error condition. 

Is the issue an artifact of how I'm constructing the MultiReaderIterator from the Encoder?